### PR TITLE
Integrate Sync stack for Metrics advisor

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/resources/session-records/DataFeedWithCredentialsTest.testBlobStorage[1].json [hunk 1]
```diff
@@ -4,103 +4,108 @@
     "Uri" : "https://REDACTED.cognitiveservices.azure.com/metricsadvisor/v1.0/dataFeeds",
     "Headers" : {
       "User-Agent" : "azsdk-java-azure-ai-metricsadvisor/1.2.0-beta.1 (11.0.9; Windows 10; 10.0)",
-      "x-ms-client-request-id" : "4dcc1081-0946-4f80-986c-b0f236d4b7f2",
+      "x-ms-client-request-id" : "003a1971-e066-4ac7-8c72-068cd64765ee",
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/resources/session-records/DataFeedWithCredentialsTest.testBlobStorage[1].json:7]: Why did we need to re-record some tests ?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/util/Utility.java [hunk 2]
```diff
@@ -39,4 +70,187 @@ public static String parseOperationId(String operationLocation) {
     public static String toStringOrNull(Object obj) {
         return obj != null ? obj.toString() : null;
     }
+
+    public static Context enableSync(Context context) {
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/implementation/util/Utility.java:74]: with Liudmila's latest change, this should not be required anymore.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/resources/session-records/DataFeedWithCredentialsTest.testBlobStorage[1].json (hunk 1 above)
Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/resources/session-records/DataFeedWithCredentialsTest.testBlobStorage[1].json:7]: I think earlier, there was a bug with one of the API's. deleteDataSourceCredential was earlier routing to deletDatafeed. The updated recordings are now doing the correct thing.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/DataFeedClientTest.java [hunk 3]
```diff
@@ -197,7 +197,7 @@ void testListDataFeedFilterByCreator(HttpClient httpClient, MetricsAdvisorServic
     // @MethodSource("com.azure.ai.metricsadvisor.TestUtils#getTestParameters")
     // void testListDataFeedSkip(HttpClient httpClient, MetricsAdvisorServiceVersion serviceVersion) {
     //     // Arrange
-    //     client = getMetricsAdvisorAdministrationBuilder(httpClient, serviceVersion).buildClient();
+    //     client = getMetricsAdvisorAdministrationBuilder(httpClient, serviceVersion, true).buildClient();
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/DataFeedClientTest.java:200]: should this test be enabled/un- commented or removed ?

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java [hunk 4]
```diff
@@ -22,9 +23,29 @@ public abstract class MetricsAdvisorAdministrationClientTestBase extends TestBas
     protected void beforeTest() {
     }
 
+    private HttpClient buildAsyncAssertingClient(HttpClient httpClient) {
+        return new AssertingHttpClientBuilder(httpClient)
+            .skipRequest((ignored1, ignored2) -> false)
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java:28]: the skipRequest call has this as default value I think now, it doesn't need to be specified.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java [hunk 5]
```diff
@@ -22,9 +23,29 @@ public abstract class MetricsAdvisorAdministrationClientTestBase extends TestBas
...
+            .build();
+    }
+
     MetricsAdvisorAdministrationClientBuilder getMetricsAdvisorAdministrationBuilder(HttpClient httpClient,
-        MetricsAdvisorServiceVersion serviceVersion) {
-        return getMetricsAdvisorAdministrationBuilder(httpClient, serviceVersion, true);
+                                                         MetricsAdvisorServiceVersion serviceVersion, boolean isSync) {
+        HttpClient httpClient1;
+        if (isSync) {
+            httpClient1 = buildSyncAssertingClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java:44]: nit: consider handling null http client input when test mode isn't playback.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorClientTestBase.java [hunk 6]
```diff
@@ -20,12 +21,32 @@ public abstract class MetricsAdvisorClientTestBase extends TestBase {
...
+            .skipRequest((ignored1, ignored2) -> false)
+            .assertSync()
+            .build();
     }
 
     MetricsAdvisorClientBuilder getMetricsAdvisorBuilder(HttpClient httpClient,
+                                                         MetricsAdvisorServiceVersion serviceVersion, boolean isSync) {
+        HttpClient httpClient1;
+        if (isSync) {
+            httpClient1 = buildSyncAssertingClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
```
Comment (g2vinay) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorClientTestBase.java:42]: nit: consider handling null http client input when test mode isn't playback.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/DataFeedClientTest.java (hunk 3 above)
Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/DataFeedClientTest.java:200]: yeah, there is an open issue to fix this https://github.com/Azure/azure-sdk-for-java/issues/33586

File: sdk/metricsadvisor/azure-ai-metricsadvisor/swagger/README.md [hunk 7]
```diff
@@ -35,6 +35,8 @@ context-client-method-parameter: true
 custom-types-subpackage: models
 custom-types: AnomalyStatus,AnomalyValue,ChangePointValue,EnrichmentStatus,FeedbackType,AnomalyIncidentStatus,PeriodType,AnomalySeverity,AlertQueryTimeMode,MetricSeriesDefinition,FeedbackQueryTimeMode,AnomalyAlert,DataFeedGranularityType,DataFeedRollupType,DataFeedAutoRollUpMethod,DataFeedStatus,MetricsAdvisorErrorCodeException,MetricsAdvisorErrorCode
 default-http-exception-type: com.azure.ai.metricsadvisor.models.MetricsAdvisorResponseException
+enable-sync-stack: true
+polling: {}
```
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/swagger/README.md:39]: https://github.com/Azure/autorest.java#polling-configuration

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java (hunk 5 above)
Comment (samvaity) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/MetricsAdvisorAdministrationClientTestBase.java:44]: httpClient won't be null when not in playback?

**Pull request merged**